scripts/etc/init.d/rcS

```diff
@@ -301,40 +301,43 @@ esac
 # configure different kinds of presets
 if [ "$cdebootstrap_cmdline" = "" ]; then
 
-    # minimal
-    cdebootstrap_cmdline="--flavour=minimal --include=kmod,fake-hwclock,ifupdown,net-tools,ntp,openssh-server$dhcp_packages"
+    # from small to large: base, minimal, server
+    # not very logical that minimal > base, but that's how it was historically defined
 
-    # add dhcp client if using dhcp
-    if [ "$ip_addr" = "dhcp" ]; then
-        cdebootstrap_cmdline="$cdebootstrap_cmdline,isc-dhcp-client"
-    fi
+    # base
+    base_packages="${kernel_meta_package},cpufrequtils,kmod"
 
-    # add latest kernel if rootfstype requires a module. 
-    # Changed it so it is always installed.
-    # Firmware can't be installed just yet, we'll do so at the end
-    if [ "$kernel_module" = true ] ; then
-        cdebootstrap_cmdline="$cdebootstrap_cmdline,${kernel_meta_package}"
-    fi
+    # minimal
+    minimal_packages="fake-hwclock,ifupdown,net-tools,ntp,openssh-server"
+
+    # server
+    server_packages="vim-tiny,iputils-ping,wget,ca-certificates,rsyslog,cron,dialog,locales,less,man-db"
 
     case $preset in
-        minimal)
-            ;;
         base)
-            cdebootstrap_cmdline="--flavour=minimal"
+            cdebootstrap_cmdline="--flavour=minimal --include=${base_packages}"
+            ;;
+        minimal)
+            cdebootstrap_cmdline="--flavour=minimal --include=${base_packages},${minimal_packages}"
             ;;
         *)
-            cdebootstrap_cmdline="$cdebootstrap_cmdline,vim-tiny,iputils-ping,wget,ca-certificates,rsyslog,cron,dialog,locales,less,man-db"
-            preset=server
+            # this should be 'server', but using '*' for backward-compatibility
+            cdebootstrap_cmdline="--flavour=minimal --include=${base_packages},${minimal_packages},${server_packages}"
+            if [ "$preset" != "server" ]; then
+                echo "Unknown preset specified: $preset"
+            fi
             ;;
     esac
 
-    # add extra packages
+    dhcp_client_package="isc-dhcp-client"
+    # add dhcp client if using dhcp
+    if [ "$ip_addr" = "dhcp" ]; then
```

Comment:
should we play it safe and assume dhcp if ```${id_addr}``` is empty or even not set? (ok, not set isn't going to happen, I admit)